lasagne/layers/merge.py

```diff
@@ -91,7 +128,7 @@ def get_output_for(self, inputs, **kwargs):
             if coeff != 1:
                 input *= coeff
             if output is not None:
-                output += input
+                output = self.merge_function(output, input)
             else:
                 output = input
         return output
```

Comment:
Okay, maybe I'm more used to reading list comprehensions, but the main point is not length, but keeping the implementations independent. Now you're coupling the subclass implementation to the super class implementation naming its attribute `merge_function`. It's not a big deal, but the software engineer in me cringes!
What about:
```python
def get_output_for(self, inputs, **kwargs):
    inputs = list(inputs)
    for idx in range(len(inputs)):
        if self.coeffs[idx] != 1:
            inputs[idx] = inputs[idx] * self.coeffs[idx]
    return super(ElemwiseSumLayer, inputs, **kwargs)
```
Although I'd really prefer:
```python
def get_output_for(self, inputs, **kwargs):
    # multiply each input by its coefficient if needed
    inputs = [input * coeff if coeff != 1 else input
              for coeff, input in zip(self.coeffs, inputs)]
    # pass inputs to the super class for summing
    return super(ElemwiseSumLayer, inputs, **kwargs)
```